### ISSUE TITLE
Sheets can lose their scroll bar (Filters Help)

### ISSUE THREAD

**Issue by robojumper** [DestinyItemManager/DIM#10346]
### DIM Version

Version 8.16.0.3284 (beta), built on 4/16/2024, 10:58:13 PM

### Browser Details

Edge 123, Firefox 125

### OS Details

Windows 10

### Describe the bug

* Open the Filters Help sheet from the search bar
* As expected, the sheet has a scroll bar that allows scrolling to view all filters
* Type `123` in the sheet search bar, which doesn't match any filters
* Scroll bar disappears
* Clear sheet search bar contents
* Unexpectedly, scroll bar is still hidden, which prevents scrolling to view filters

### Logs

```shell
None relevant
```

**Comment by bhollis** [comment]
Related to #9240, caused by sheets recalculating their maximum height incorrectly

**Comment by ryan-rushton** [comment]
Is the issue that the sheet is calculating the max height but loading contents is async? If that is the case we need to provide the contents a callback it can call when its contents have loaded correctly.

**Comment by ryan-rushton** [comment]
@robojumper can you test and close for me? It should be live on beta now.

**Comment by robojumper** [comment]
I'm still seeing this with given repro steps in Version 8.19.0.3324 (beta)

**Comment by ryan-rushton** [comment]
Hmmm ok, it must be something else. It fixed the other issue Ben linked. I will have a look.

**Comment by ryan-rushton** [comment]
@robojumper I cannot reproduce this on chrome or firefox (125 video attached). Could this be a windows thing?

https://github.com/DestinyItemManager/DIM/assets/7344652/db42f3bd-21cc-4558-ba0d-b9f1045eb22e

**Comment by robojumper** [comment]
This affects the desktop layout for me. You might be able to reproduce it with the dev tools mobile emulation turned off?

**Comment by ryan-rushton** [comment]
I tried both, which are both fine for me. I will get a vid of firefox there too

**Comment by ryan-rushton** [comment]
https://github.com/DestinyItemManager/DIM/assets/7344652/1a1604d2-06b4-493e-88a4-562beac459d1
<img width="452" alt="Screenshot 2024-05-09 at 6 58 08 PM" src="https://github.com/DestinyItemManager/DIM/assets/7344652/8649316f-6a43-4789-8a5e-7587aa8a5ba5">

**Comment by ryan-rushton** [comment]
I'll try reproduce this on my gaming PC when I get a chance and remember to.

**Comment by ryan-rushton** [comment]
Just tried this again on Firefox 126 and 125.03 on windows and I don't see the issue. @robojumper do you have any non default settings enabled in firefox?

**Comment by robojumper** [comment]
My Firefox is pretty customized but it also happens in Edge 
![GIF 19 05 2024 16-12-45](https://github.com/DestinyItemManager/DIM/assets/14299449/fd4dd91e-ebdf-400e-b8b1-7ddad054a988)

**Comment by ryan-rushton** [comment]
Ok, I will try with that next.

**Comment by bhollis** [comment]
I wasn't able to repro on Chrome or Firefox for macOS.

**Comment by ryan-rushton** [comment]
Yeah I have tried a few combo's and can't repro. I need to do the edge one but I think this is due to something non-default in robo's settings.

**Comment by delphiactual** [comment]
as soon as i type the `d` in `asd` i lose the scroll bar on chrome (windows)

**Comment by ryan-rushton** [comment]
And to be clear @delphiactual it doesn't come back once deleting everything?

Also, what version of windows? I am on 11 and note that robo is on 10. So if you are on 10 maybe its a 10 thing

**Comment by delphiactual** [comment]
i am on win 11, and it does not come back

**Comment by ryan-rushton** [comment]
Managed to reproduce on chrome (I swear I checked that initially). PTAL at the fix if you get a chance

**Comment by ryan-rushton** [comment]
This also seems related to https://github.com/DestinyItemManager/DIM/issues/9886 potentially? There seems to be some weird behaviour with this useResizerObserver hook
